ui/common/src/socket.ts

```diff
@@ -289,57 +283,65 @@ export default class StrongSocket {
     const ws = this.ws;
     if (ws) {
       this.debug('Disconnect');
-      this.autoReconnect = false;
       ws.onerror = ws.onclose = ws.onopen = ws.onmessage = () => {};
       ws.close();
     }
   };
 
-  onError = (e: unknown): void => {
+  private onError = (e: unknown): void => {
+    if (this.heartbeat.wasSuspended) return;
     this.options.debug = true;
     this.debug(`error: ${e} ${JSON.stringify(e)}`); // e not always from lila
   };
 
-  onClose = (e: CloseEvent, url: string): void => {
+  private onClose = (e: CloseEvent): void => {
     site.pubsub.emit('socket.close');
-    if (this.autoReconnect) {
+
+    if (this.heartbeat.wasSuspended) return this.onSuspended();
+    this.storeStats({ event: 'close', code: e.code });
+
+    if (this.ws) {
       this.debug('Will autoreconnect in ' + this.options.autoReconnectDelay);
       this.scheduleConnect(this.options.autoReconnectDelay);
     }
     if (e.wasClean && e.code < 1002) return;
 
-    if (isOnline())
-      log(`${site?.sri ? 'sri ' + site.sri : ''} unclean close ${e.code} ${url} ${e.reason}`);
-    this.tryOtherUrl = true;
+    if (isOnline()) this.tryOtherUrl = true;
     clearTimeout(this.pingSchedule);
   };
 
-  onSuccess = (): void => {
-    this.nbConnects++;
```

Comment:
`this.nbConnects` are related check are redundant with the one time event hasConnected